Port fix for a segfault

stefan11111 opened the pull request:
This fixes a segfault in the kdrive mouse driver.
I haven't managed to get tinyx to segfault, because the ability to pass mouse protocol by command line args seems to not be implemented.
Still, this doesn't mean that the segfault can't happen here too. It doesn't happen on my system because the default ps/2 protocol works fine for me. However, this might not be the case on other systems.

This code fixes an incorrect loop condition and adds a fallback for unknown mouse protocols.
Also replaces strlen with sizeof appropriately.

stefan11111 committed:
port fix for a segfault: https://github.com/stefan11111/kdrive/commit/0aa5ca446ec61aebcf6c10ac2f00ec61cd53cf4a

This fixes an incorrect loop condtion, as it is possible to skip over NUM_PROT when incrementing
This causes a segfault in kdrive, when km->i_prot would have to loop, instead continuing to read out of bounds.
I couldn't get tinyx to also segfault, but that doesn't mean it's impossible. It might happen on different hardware.

rofl0r commented on kdrive/linux/mouse.c:
```diff
@@ -386,8 +386,7 @@ static const KmouseProt exps2Prot = {
 static const unsigned char ps2_init[] = {
-	PSMC_ENABLE_DEV,
-	0,
+	PSMC_ENABLE_DEV
```
how does removing these 0's fix a segfault ?

rofl0r commented:
this commit changes 4 things, but none of the changes are explained.
imagine you stumble upon this commit without having authored it and see the "==" changed to ">=" and wonder what was wrong. or why strlen was changed to sizeof. the commit message has plenty of space (in fact almost unlimited) to explain each and every change. even better would be if each change was a separate commit with an explanation.

stefan11111 commented:
> this commit changes 4 things, but none of the changes are explained.
> imagine you stumble upon this commit without having authored it and see the "==" changed to ">=" and wonder what was wrong. or why strlen was changed to sizeof. the commit message has plenty of space (in fact almost unlimited) to explain each and every change. even better would be if each change was a separate commit with an explanation.

I can make separate commits. I assumed you would prefer having all changes in a single commit.

stefan11111 committed:
Replace strlen with sizeof

We know the size of the ps/2 init bytes at compile time. No need to call strlen.

stefan11111 commented:
> even better would be if each change was a separate commit with an explanation.

Done.
